src/lib/components/explorer/export/ExportStepper.svelte

```diff
@@ -1,609 +1,228 @@
+        .map((info) => ({ resourceId: info?.resourceId, name: info?.name, queryId: info?.queryId }))
+        .filter((v) => v.queryId);
+
+      saveDatasetPromise = createDatasetName(
+        datasetId,
+        datasetNameInput,
+        siteQueryIds.length > 0 ? siteQueryIds : undefined,
+      ).then((data: DataSet) => {
+        if (features.federated) {
+          statusPromise = Promise.resolve();
+          return data;
+        } else {
+          statusPromise = checkExportStatus(picsureResultId);
+        }
         return data;
       });
     }
+    return;
   }
 
-  async function submitQuery(): Promise<void> {
-    let interval: NodeJS.Timeout;
-    const statetext = {
-      initial: 'Creating dataset ID...',
-      waiting: 'Hang tight. We are still working on it...',
-      retry: "Something's taking longer than usual. We are still working on it...",
-    };
-
-    function requestUpdate(method: () => Promise<void | DataSetResponse>, retry: boolean = true) {
-      processingMessage = retry ? statetext.initial : statetext.retry;
-      if (retry) {
-        interval = setInterval(() => {
-          processingMessage = statetext.waiting;
-        }, PROMISE_WAIT_INTERVAL * 1000);
-      }
-      datasetIdPromise = method()
-        .finally(() => clearInterval(interval))
-        .catch((err) => {
-          if (retry) {
-            requestUpdate(method, false);
-          } else {
-            throw err;
-          }
-        });
-    }
-
-    try {
-      query.query.fields = $exports.map((exp) => exp.conceptPath);
-      query.query.expectedResultType = activeType || 'DATAFRAME';
-      datasetId = '';
-      requestUpdate(() =>
-        api.post(Picsure.Query, query).then((res: DataSetResponse) => {
-          datasetId = res.picsureResultId || 'Error';
-        }),
-      );
-      await datasetIdPromise;
-    } catch (error) {
-      $stepperState.current--;
-      console.error('Error in submitQuery', error);
-      throw error;
-    }
-  }
-
-  function checkExportStatus(lastPicsureResultId?: string) {
+  async function checkExportStatus(lastPicsureResultId?: string) {
     const statusId = lastPicsureResultId || datasetId;
+    const queryFragment = `/${statusId}/status`;
     return api
-      .post(`${Picsure.Query}/${statusId}/status`, query)
-      .then((res: { picsureResultId: string; status: string }) => {
-        if (res.status === 'ERROR') {
-          lockDownload = true;
-          return Promise.reject(res.status);
-        }
-        picsureResultId = res.picsureResultId;
-        lockDownload = false;
-      });
+      .post(`${Picsure.Query}${queryFragment}`, query)
+      .then(
+        (res: {
+          picsureResultId: string;
+          status: string;
+          resultMetadata: { picsureQueryId: string };
+        }) => {
+          if (res.status === 'ERROR') {
+            lockDownload = true;
+            return Promise.reject(res.status);
```

Comment:
Maybe add a catch here in case of error - I don't see that we're handling it atm.